### PR TITLE
Fix bug with accepting multiple items in an xml list

### PR DESCRIPTION
**Pull request by TimboTambo**
# Fix bug with accepting multiple items in an xml list

Allow multiple items in a list by restructuring xml structs.

### Acceptance Criteria

- [ ] Accepts xml request with multiple items

### Testing information

Make request with body:
`<document>
  <id>con33333333</id>
  <name>Name of an SPC</name>
  <type>SPC</type>
  <author>theauthor</author>
  <products>
    <product>This is a product</product>
    <product>This is another product</product>
  </products>
  <pl_number>PL 12345/0010-0001</pl_number>
  <active_substances>
    <active_substance>Caffeine</active_substance>
    <active_substance>Paracetamol</active_substance>
  </active_substances>
  <keywords>
    <keyword>Keyword 1</keyword>
    <keyword>Keyword 2</keyword>
  </keywords>
  <file_source>Sentinel</file_source>
  <file_path>example_file.txt</file_path>
</document>`

### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/doc-index-updater/src/models.rs [hunk 1]
```diff
@@ -83,22 +83,25 @@ impl Into<Document> for XMLDocument {
...
                 ),
                 None => None,
             },
             pl_number: self.pl_number,
             active_substances: self
                 .active_substances
+                .active_substance
                 .iter()
-                .map(move |active_substance| active_substance.name.clone())
+                .map(move |active_substance| active_substance.clone())
```
Comment (sgrowe) [inline medicines/doc-index-updater/src/models.rs]: I think the `move`s in all of these closures are unnecessary?

File: medicines/doc-index-updater/src/models.rs [hunk 2]
```diff
@@ -83,22 +83,25 @@ impl Into<Document> for XMLDocument {
             author: self.author,
             products: self
                 .products
+                .product
                 .iter()
-                .map(move |active_substance| active_substance.name.clone())
+                .map(|product| product.to_string())
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/models.rs]: This 3-line block is unnecessary. You're iterating just to return the initial thing.

File: medicines/doc-index-updater/src/models.rs [hunk 3]
```diff
@@ -83,22 +83,25 @@ impl Into<Document> for XMLDocument {
...
                 .collect::<Vec<String>>(),
             keywords: match self.keywords {
                 Some(kw) => Some(
-                    kw.iter()
-                        .map(move |keyword| keyword.name.clone())
+                    kw.keyword
+                        .iter()
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/models.rs]: This 3-line block is unnecessary. You're iterating just to return the initial thing.

File: medicines/doc-index-updater/src/models.rs [hunk 4]
```diff
@@ -83,22 +83,25 @@ impl Into<Document> for XMLDocument {
...
                 ),
                 None => None,
             },
             pl_number: self.pl_number,
             active_substances: self
                 .active_substances
+                .active_substance
                 .iter()
-                .map(move |active_substance| active_substance.name.clone())
+                .map(|active_substance| active_substance.to_string())
```
Comment (rjkerrison) [inline medicines/doc-index-updater/src/models.rs]: This 3-line block is unnecessary. You're iterating just to return the initial thing.

File: medicines/doc-index-updater/src/models.rs (hunk 2 above)
Comment (m-doughty) [reply medicines/doc-index-updater/src/models.rs]: Yeah these blocks are no longer necessary, they were necessary when we weren't already deserializing to `Vec<String>`.

File: medicines/doc-index-updater/src/models.rs (hunk 4 above)
Comment (TimboTambo) [reply medicines/doc-index-updater/src/models.rs]: It's my birthday and I'll iterate if I want to!

**Pull request merged**